### DISCUSSION TITLE
Lets finally get rid of jk

### DISCUSSION THREAD

**Discussion by favetelinguis** [meow-edit/meow#599] [Show and tell]
Using jk for navigation has long been a bad idea it feels, using hjkl has stoped me from expanding my knowledge about VIM keys (been using it for 10years but its amazing how far you can get with basic understanding). I have used vim, evil, ideavim and never really cared about trying to make any adjustments. Over the years I have navigated more and more towards emacs since its always feels like the best option and i do like Lisp. 
Using evil has always felt like an ok compromise, I did not understand anything about what happened and I did know that I sacrificed much of emacs magic but hey I got to use hjkl and that made me happy. After discovering meow my eyes opened up and I started to feel that it was the answer to all my question (I have used kakoune and I do think it is a better experience then VIM but still not enough to get me away from Lisp, and helix is just naw).

And now finally to the point, given how easy it is to redefine and understand meow I decided to get rid of using jk since that never is a good match in the emacs landscape and constantly will destroy the feeling in move-mode compared to vanilla emacs. I want to share below some of the basic modifications I have done to see if anyone else has tried the same or something similar. I do understand that this will handicap my muscle memory when switching to vim or ideavim etc but I just dont care any longer (im 45+ yo).

```
  (defun my/remove-current-mark ()
    "Remove the current mark from the mark ring and the global mark ring."
    (interactive)
    (when (mark)
      (let ((current-mark (mark-marker)))
        ;; Deactivate the current mark
        (deactivate-mark)
        ;; Remove the current mark from the local mark ring
        (setq mark-ring (delete current-mark mark-ring))
        ;; Remove the current mark from the global mark ring
        (setq global-mark-ring (delete current-mark global-mark-ring))
        ;; Ensure the current mark is not set
        (set-marker current-mark nil))))

  (defun my/jumper (&optional arg)
    (interactive)
    (if (region-active-p)
	(meow-search arg)
      (avy-goto-char-timer)))

--- Change in motion
     '("J" . pop-global-mark)
     '("K" . my/remove-current-mark)
     '("n" . meow-next)
     '("p" . meow-prev)
--- Change in normal
     '("j" . my/jumper)
     '("J" . pop-global-mark)
     '("k" . meow-kill)
     '("K" . my/remove-current-mark)
     '("s" . meow-yank)
     '("S" . meow-clipboard-yank)
     '("n" . meow-next)
     '("p" . meow-prev)
```

My reasoning for thinking this makes sense is as follow. Using np for movement in emacs is a no brainier, if you are ok with accepting that, many things get simpler, you almost never get a conflict in motion-mode since np are usually down/up anyways.

Now what is jk its usually jump and kill. So with some custom function I am using j for moving in the buffer, if there is a selection it will work as n in meow and if there is no selection i get the avy super power move. J if for cycling the global mark ring to go back to places I want to remember. k is for killing in buffer, it does what s did in the original binding, K is for removing marks, this way I can clean up places I no longer want to go back to. I have found this to be a very basic but efficient way of moving in buffer and between buffers. 

Please give this a try and see what you think, just giving up jk and realizing that if you use jk for navigation you are not gaining much from a powerful mode as vim/meow.

**Comment by DogLooksGood** [comment]
You don't have to specify n/p for next/prev. It's usually the default behavior in most special modes.

`meow-prev` and `meow-next` will use the commands on `C-p` and `C-n`, sometimes they are different from the default `p/n` in the special mode.
